models/maker/maker_accounting.sql

```diff
@@ -0,0 +1,861 @@
+(
+    SELECT ilk, call_block_time, dink
+    FROM yielding_psms_trxns
+    
+    UNION ALL
+    
+    SELECT ilk, period, amt
+    FROM yielding_psms_cart_ilk
+), yielding_psms_daily_agg AS
+(
+    SELECT DATE_TRUNC('day',call_block_time) AS period
+    , ilk
+    , SUM(dink)/POW(10,18) AS amt
+    FROM yielding_psms_unioned
+    GROUP BY 1,2
+), yielding_psms_balance AS 
+(
+    SELECT period
+    , ilk
+    , SUM(amt) OVER (PARTITION BY ilk ORDER BY period ASC) AS balance
+    FROM yielding_psms_daily_agg
+), yielding_psms_monthly_avg AS 
+(
+    SELECT LEFT(CAST(period AS STRING),7) AS period
+    , YEAR(period) AS year
+    , MONTH(period) AS month
+    , ilk
+    , AVG(balance) OVER (PARTITION BY ilk, LEFT(CAST(period AS STRING),7)) AS avg_balance
+    , COUNT(balance) OVER (PARTITION BY ilk, LEFT(CAST(period AS STRING),7)) AS num_days
+    FROM yielding_psms_balance
+    QUALIFY ROW_NUMBER() OVER (PARTITION BY ilk, LEFT(CAST(period AS STRING),7) ORDER BY period, ilk) = 1 --1 row per month taken post averaging
+), yielding_psms AS
+(
+    SELECT year
+    , month
+    , revenue_code AS code
+    , avg_balance * apr * num_days/365 AS value
+    FROM yielding_psms_monthly_avg
+    JOIN ilk_list_labeled
+    ON yielding_psms_monthly_avg.ilk = ilk_list_labeled.ilk
+    AND yielding_psms_monthly_avg.period BETWEEN COALESCE(ilk_list_labeled.begin_period, '2000-01') AND COALESCE(ilk_list_labeled.end_period, '2222-12') --if null, ensure its not restrictive
+    AND ilk_list_labeled.revenue_code = 4411
+)
+*/
+SELECT coa.code
+     , unioned.ts
+     , unioned.hash
+     , unioned.value
+     , unioned.token
+     , unioned.descriptor
+     , unioned.ilk
+FROM chart_of_accounts coa
+
+LEFT JOIN 
+(
+    SELECT ts, hash, code, value, 'DAI' AS token, 'Returned Workforce Expenses' AS descriptor, NULL AS ilk
+    FROM team_dai_burns
+    WHERE ts BETWEEN {{start_dt}} AND DATEADD({{end_dt}}, 1)
+    UNION ALL
+    SELECT ts, hash, code, value, token, descriptor, ilk
+    FROM {{ ref('maker_accounting_liquidation') }}
+    WHERE ts BETWEEN {{start_dt}} AND DATEADD({{end_dt}}, 1)
+    UNION ALL
+    SELECT ts, hash, code, value, 'DAI' AS token, 'Trading Revenues' AS descriptor, ilk
+    FROM trading_revenues
+    WHERE ts BETWEEN {{start_dt}} AND DATEADD({{end_dt}}, 1)
+    UNION ALL
+    SELECT ts, hash, code, value, 'DAI' AS token, 'MKR Mints' AS descriptor, NULL AS ilk
+    FROM mkr_mints
+    WHERE ts BETWEEN {{start_dt}} AND DATEADD({{end_dt}}, 1)
+    UNION ALL
+    SELECT ts, hash, code, value, 'DAI' AS token, 'MKR Burns' AS descriptor, NULL AS ilk
+    FROM mkr_burns
+    WHERE ts BETWEEN {{start_dt}} AND DATEADD({{end_dt}}, 1)
+    UNION ALL
+    SELECT ts, hash, code, value, token, descriptor, ilk
+    FROM {{ ref('maker_accounting_interest') }}
+    WHERE ts BETWEEN {{start_dt}} AND DATEADD({{end_dt}}, 1)
+    UNION ALL
+    SELECT ts, hash, code, value, 'DAI' AS token, 'OpEx' AS descriptor, NULL AS ilk
+    FROM opex
```

Comment:
Sorry for the unclear statement. Since this is a long query, we can take the CTE out from this file to other individual files(for easier maintenance and future development). I have make `{{ ref('maker_accounting_liquidation') }}` and `{{ ref('maker_accounting_interest') }}` as example. We can also add test for these individual queries.

Reply:
I agree on having subqueries, this is a great idea and we are already working like that on ENS quaccounting queries.

Could you commit the current version where you are at, with maker_accounting_liquidation and maker_accounting_interest and @web3lyt will work on a better refactoring going forward?

I really feel that putting the maker_accounting table in action will also lead to some unexpected changes. It is an innovative and iterative endeavor!

Reply:
- Liquidation: https://github.com/duneanalytics/spellbook/pull/2415/files#diff-de1172085fb4f56b4bf4ae9f54c1b413644c2bde5118d8fc978f2abb34bb16b5
- Interest: https://github.com/duneanalytics/spellbook/pull/2415/files#diff-b8dbe62e83c1e648ef0ae8680d496b2fc799fe03ec2a95b5398707bcf4beb7b5

These codes are the same as the original CTEs.